include-blang/cuda_cooperative_groups.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CUDA_COOPERATIVE_GROUPS_H
+#define CUDA_COOPERATIVE_GROUPS_H
+
+namespace cooperative_groups {
+  class grid_group {
+    public:
+      __inline__ __device__ void sync() const {
+          bugle_grid_barrier();
+      }
+  };
+  
+  class thread_block {
+    public:
+      __inline__ __device__ void sync() const {
+          bugle_barrier(true, true);
+      }
+  };
+  
+  __device__ grid_group this_grid();
+  
+  __device__ thread_block this_thread_block();
+  
+  __inline__ __device__ void synchronize(grid_group g) {
```

Comment:
```suggestion
  static __inline__ __device__ void synchronize(grid_group g) {
```

Reply:
thank you :) I was trying to figure out how to fix that test case.